src/Pelagos/Bundle/AppBundle/Resources/public/js/data-discovery.js

```diff
@@ -99,8 +106,213 @@ function resizeLeftRight() {
+
+
+    if (!data["datasetSubmission"]["authors"]) {
+        $(rowContent).find("#container-dataset-authors").hide();
+    } else {
+        $(rowContent).find("#dataset-authors").text(data["datasetSubmission"]["authors"]);
+    }
+
+    if (!data["year"]) {
+        $(rowContent).find("#container-dataset-year").hide();
+    } else {
+        $(rowContent).find("#dataset-year").text(data["year"]);
+    }
+
+    $(rowContent).find("#dataset-title").attr("href", $(rowContent).find("#dataset-title").attr("href").replace("placeholder-udi", data["udi"]) );
+    $(rowContent).find("#dataset-title").text(data["title"]);
+
+    if (!data["researchGroup"]["name"]) {
+        $(rowContent).find("#container-dataset-research-group").hide();
+    } else {
+        $(rowContent).find("#dataset-research-group").text(data["researchGroup"]["name"]);
+    }
+
+    if (!data["doi"]["doi"]) {
+        $(rowContent).find("#container-dataset-doi").hide();
+    } else {
+        $(rowContent).find("#dataset-doi").attr("href", $(rowContent).find("#dataset-doi").attr("href").replace("placeholder-doi", data["doi"]["doi"]));
+        $(rowContent).find("#dataset-doi").text(data["doi"]["doi"]);
+    }
+
+    var filesize = parseInt(data["datasetSubmission"]["datasetFileSize"]);
+    if (filesize) {
+        if (filesize > 1073741824) {
+            $(rowContent).find("#dataset-filesize").css("color","red");
+        }
+        $(rowContent).find("#dataset-filesize").text(formatBytes(filesize,0));
+    }
+
+    //this shows the details dataset on Show Details button
+    $(rowContent).find(".details_link").bind("click", function(){
+        var row = $(this).closest("tr");
+        if ($(row).has(".details:empty").length == 1) {
+            var datasetId = $(row).attr("datasetid");
+            $.ajax({
+                "url": Routing.generate("pelagos_app_ui_datadiscovery_showdetails", { "id": datasetId }),
+                "success": function(data) {
+                    $(row).find('.details').html(data);
+                    $(row).find('.details').show();
+                    $(row).find('.details_link').html("Hide Details");
+                }
+            });
+        }
+        else {
+            if ($(row).find('.details:visible').length == 1) {
+                $(row).find('.details').hide();
+                $(row).find('.details_link').html("Show Details");
+            }
+            else {
+                $(row).find('.details').show();
+                $(row).find('.details_link').html("Hide Details");
+            }
+        }
+    });
+
+    return rowContent;
+}
+//end createRow
+
+//this occurs at initial/new search
 function showDatasets(by,id) {
+    var filter = jQuery("#filter-applied").val();
+    geo_filter = "";
+    if (trees["tree"].geo_filter) {
+        geo_filter = trees["tree"].geo_filter;
+    }
+
+    //reset existing data to prepare for a new search
+    currentIndex = 0;
+    datasetList[0] = [];
```

Comment:
Similarly, even here you can just reset the whole array `datasetList = []`

Reply:
If I don't explicitly initialize/reset them as a new list/array in here it would break the page (seems to be JS thing)

Reply:
Yeah, I tried it too.. I'm not really sure how the global variable is working. It should also throw the same undefined error, but its not. Seems weird.